src/pages/Applications/Filter/schema.ts

```diff
@@ -0,0 +1,28 @@
+import { ObjectSchema, date, lazy, object, ref, string } from "yup";
+import { FormValues } from "./types";
+import { SchemaShape } from "schemas/types";
+
+const endKey: keyof FormValues = "endDate";
+const startKey: keyof FormValues = "startDate";
+
+const now = new Date();
+const dateSchema = date()
+  .typeError("invalid")
+  .max(now, "can't be later than today");
+
+export const schema = object<any, SchemaShape<FormValues>>({
+  startDate: dateSchema.when(endKey, ([endDate], schema) =>
+    date().isValidSync(endDate)
+      ? schema.max(endDate, "can't be later than end date")
+      : schema
+  ),
+  endDate: lazy((val) =>
+    val
+      ? dateSchema.when(startKey, ([startDate], schema) =>
+          date().isValidSync(startDate)
+            ? schema.min(ref(startKey), "can't be earlier than start date")
+            : schema
+        )
+      : string().required("invalid")
```

Comment:
```suggestion
      : requiredString
```
nit: since we have the wrapper

Reply:
reason for not using that is it's error is `required`, dates are a bit tricky to work with, clearing the date by `clear` button resolves to `""` but clearing it with keyboard section by section leads to invalid date - decided to just generalize to `invalid`